### PR TITLE
Clarifying icon usage in standalone link

### PR DESCRIPTION
**Pull request by heatherlarsen**
### :pushpin: Summary

If merged, this PR adds more guidance around when to use a leading or trailing icon in the standalone link component.

👉🏻 **Preview link:** https://hds-website-git-heatherlarsen-link-icon-usage-hashicorp.vercel.app/components/link/standalone#icon-position

### :link: External links

Jira ticket: [HDS-2216](https://hashicorp.atlassian.net/browse/HDS-2216)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed
- [ ] Confirm that A11y tests have been run locally for this component

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2216]: https://hashicorp.atlassian.net/browse/HDS-2216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jorytindall — APPROVED** [review]
:shipit:

File: website/docs/components/link/standalone/partials/guidelines/guidelines.md [hunk 1]
```diff
@@ -52,41 +52,84 @@ A Standalone Link can include a leading or trailing icon. Avoid creating links w
...
 
-Use the leading position when using service icons (e.g., GitHub).
+!!! Do
+
+Consider directionality; if directing the user backward in a flow, use the leading position. 
+
+<Doc::Layout @spacing="12px">
+  <Hds::Link::Standalone @text="Back to Dashboard" @icon="chevron-left" @iconPosition="leading" @color="primary" @route="components" />
+  <Hds::Link::Standalone @text="Back to Dashboard" @icon="chevron-left" @iconPosition="leading" @color="secondary" @route="components" />
+</Doc::Layout>
```
Comment (didoo) [inline website/docs/components/link/standalone/partials/guidelines/guidelines.md:71]: I don't think we need to duplicate every time the example for primary and secondary color (it was confusing, for me). if you want to show that there are different variants, alternate the examples between the two colors

(this applies to other examples too)

```suggestion
<Hds::Link::Standalone @text="Back to Dashboard" @icon="chevron-left" @iconPosition="leading" @color="primary" @route="components" />
```

**Review by didoo — COMMENTED** [review]
Left a comment, mainly around duplicating the examples for the two color variants

**Review by MelSumner — APPROVED** [review]
I like it, this looks good!

File: website/docs/components/link/standalone/partials/guidelines/guidelines.md [hunk 2]
```diff
@@ -52,41 +52,84 @@ A Standalone Link can include a leading or trailing icon. Avoid creating links w
...
+  <Hds::Link::Standalone @text="Waypoint" @icon="waypoint-color" @iconPosition="leading" @color="primary" @route="components" />
 </Doc::Layout>
 
+!!!
+
+
 ### Trailing
 
-Generally, we recommend `arrow-right` for internal links and `external-link` (or similar) for external links. 
+Consider trailing icons when there's no other meaningful icon or when guiding the user forward through the product.
```
Comment (KristinLBradley) [inline website/docs/components/link/standalone/partials/guidelines/guidelines.md:93]:
```suggestion
Consider trailing icons when there’s no other meaningful icon or when guiding the user forward through the product. 
```

**Review by KristinLBradley — APPROVED** [review]

**Pull request merged**